caravel/assets/javascripts/SqlLab/reducers.js

```diff
@@ -13,13 +13,39 @@ const defaultQueryEditor = {
 
 export const initialState = {
   alerts: [],
-  queries: [],
+  queries: {},
   queryEditors: [defaultQueryEditor],
   tabHistory: [defaultQueryEditor.id],
   tables: [],
   workspaceQueries: [],
 };
 
+function removeFromDict(state, arrKey, obj, idKey = 'id') {
+  const newDict = {};
+  for (var key in state[arrKey]) {
+    if (!(key === obj[idKey])) {
+      newDict[idKey] = state[arrKey][idKey]
+    }
+  }
+  return Object.assign({}, state, { [arrKey]: newDict });
+}
+
+function addToDict(state, arrKey, obj) {
```

Comment:
@ascott 
yes. This function mimics the addToArr function.
In theory it can be any object in the redux store.